docs/architecture/adr-063-privval-grpc.md

```diff
@@ -0,0 +1,106 @@
+# ADR 063: Privval gRPC
+
+## Changelog
+
+- 23/11/2020: Initial Version (@marbar3778)
+
+## Context
+
+Validators use remote signers to help secure their keys. This system is Tendermint's recommended way to secure validators, but the path to integration with Tendermint's private validator client is plagued with custom protocols. 
+
+Tendermint uses its own custom secure connection protocol (`SecretConnection`) and a raw tcp/unix socket connection protocol. The secure connection protocol until recently was exposed to man in the middle attacks and can take longer to integrate if not using Golang. The raw tcp connection protocol is less custom, but has been causing minute issues with users. 
+
+Migrating Tendermint's private validator client to a widely adopted protocol, gRPC, will ease the current maintenance and integration burden experienced with the current protocol. 
+
+## Decision
+
+After discussing with multiple stake holders, [gRPC](https://grpc.io/) was decided on to replace the current private validator protocol. gRPC is a widely adopted protocol in the micro-service and cloud infrastructure world. gRPC uses [protocol-buffers](https://developers.google.com/protocol-buffers) to describe its services, providing a language agnostic implementation. Tendermint uses protobuf for on disk and over the wire encoding already making the integration with gRPC simpler. 
+
+## Alternative Approaches
+
+- JSON-RPC: We did not consider JSON-RPC because Tendermint uses protobuf extensively making gRPC a natural choice.
+
+## Detailed Design
+
+With the recent integration of [Protobuf](https://developers.google.com/protocol-buffers) into Tendermint the needed changes to migrate from the current private validator protocol to gRPC is not large. 
+
+The [service definition](https://grpc.io/docs/what-is-grpc/core-concepts/#service-definition) for gRPC will be defined as:
+
+```proto
+  service PrivValidatorAPI {
+    rpc GetPubKey(tendermint.proto.privval.PubKeyRequest) returns (tendermint.proto.privval.PubKeyResponse);
+    rpc SignVote(tendermint.proto.privval.SignVoteRequest) returns (tendermint.proto.privval.SignedVoteResponse);
+    rpc SignProposal(tendermint.proto.privval.SignProposalRequest) returns (tendermint.proto.privval.SignedProposalResponse);
+
```

Comment:
This treats the remote signer as the server, but we should make it the client. This way the remote signer isn't actually exposed to the network.

Reply:
It's a debatable tradeoff. It might be good to loop in more stakeholders who are working on things that might integrate with remote signers, for example IBC relayers.

Reply:
@jackzampolin do you have thoughts here? I know you recently did some work with signer servers.

Reply:
@jleni do you have thoughts here?

Reply:
We discussed this in our team standup. We like the approach where Tendermint is the client. It simplifies implementation and follows client and server architecture, where the client makes requests. The attack vectors of having the remote signer exposed to the internet can be mitigated by using two-way TLS and setting up a firewall.